subsys/usb/device_next/class/Kconfig.msc

```
@@ -30,6 +30,12 @@ config USBD_MSC_SCSI_BUFFER_SIZE
 	  Buffer size must be able to hold at least one sector. All LUNs within
 	  single instance share the SCSI buffer.
 
+config USBD_MSC_DATA_BUF_SIZE
+	int "Data Buffer size for msc data transfer"
+	default 512
```

```suggestion
	default 512 if UDC_DRIVER_HAS_HIGH_SPEED_SUPPORT
	default 64
```

would maintain default low memory usage on Full-Speed only drivers.